gufe/protocols/protocol.py

```diff
@@ -135,15 +135,16 @@ def _create(
         self,
         stateA: ChemicalSystem,
         stateB: ChemicalSystem,
-        mapping: Optional[dict[str, ComponentMapping]] = None,
+        mapping: list[ComponentMapping],
```

Comment:
Depending on the conversation below - it would be preferred if this was Optional i.e. expect None rather than a zero length list.

```suggestion
        mapping: Optional[list[ComponentMapping]],
```